flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,281 @@
+
+import logging
+from collections import MutableSet
+
+import requests
+from requests import RequestException
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+
+log = logging.getLogger('ombi_list')
+
+class OmbiSet(MutableSet):
+    supported_ids = ['imdb_id', 'tmdb_id', 'ombi_id']
+    schema = {
+        'type': 'object',
+        'properties': {
+            'base_url': {'type': 'string'},
+            'port': {'type': 'number', 'default': 3579},
+            'api_key': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'type': {'type': 'string', 'enum': ['shows', 'seasons', 'episodes', 'movies']},
+            'only_approved': {'type': 'boolean', 'default': True},
+            'include_available': {'type': 'boolean', 'default': False},
+            'include_year': {'type': 'boolean', 'default': False},
+            'include_ep_title': {'type': 'boolean', 'default': False}
+        },
+        'required': ['base_url', 'type'],
+        'additionalProperties': False
+    }
+
+    @property
+    def immutable(self):
+        return False
+
+    def __init__(self, config):
+        self.config = config
+        self._items = None
+
+    def __iter__(self):
+        return (item for item in self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    def get(self, entry):
+        return self._find_entry(entry)
+
+    def generate_series_id(self, season, episode=None):
+        tempid = 'S' + str(season.get('seasonNumber')).zfill(2)
+        if episode:
+            tempid = tempid + 'E' + str(episode.get('episodeNumber')).zfill(2)
+
+        return tempid
+
+    def generate_title(self, item, season=None, episode=None):
+        temptitle = item.get('title')
+
+        if item.get('releaseDate') and self.config.get('include_year'):
+            temptitle = temptitle + ' (' + str(item.get('releaseDate')[0:4]) +')'
+
+        if season or episode:
+            temptitle = temptitle + ' ' + self.generate_series_id(season, episode)
+            if episode:
+                if episode.get('title') and self.config.get('include_ep_title'):
+                    temptitle = temptitle + ' ' + episode.get('title')
+
+        return temptitle
+
+    def get_access_token(self):
+        parsedurl = urlparse(self.config.get('base_url'))
+        url = '%s://%s:%s%s/api/v1/Token' % (parsedurl.scheme, parsedurl.netloc, self.config.get('port'), parsedurl.path)
+        data = {'username': self.config.get('username'),
+                'password': self.config.get('password')
+        }
+        headers = { 'Content-Type': 'application/json',
+                    'Accept': 'application/json'
+        }
+        try:
+            log.debug('Logging in with username and password to get access token')
+            log.debug('URL %s', url)
```

Comment:
You should import our requests util instead https://github.com/Flexget/Flexget/blob/develop/flexget/utils/requests.py, which incidentally debug logs _all_ requests, so you can remove this debug line.